e2e-runner/common/unix/common.sh

```
@@ -91,24 +91,38 @@ execute_scripts() {
 }
 
 # Load secrets from file into environment variables
+# Sensitive keys (*_API_KEY, *_SECRET) are deferred to avoid leaking during build.
+# Call restore_deferred_secrets() before test execution.
+declare -a deferred_secret_names
+DEFERRED_SECRETS_FILE=""
+
 load_secrets() {
     if [ -n "$secretFile" ]; then
         secretFilePath="$resourcesPath/$secretFile"
         if [ -f $secretFilePath ]; then
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)